src/av/mod.rs

```diff
@@ -24,462 +24,387 @@
-    TOXAV_ERR_SEND_FRAME_FRIEND_NOT_IN_CALL,
-    /// Synchronization error occurred.
-    TOXAV_ERR_SEND_FRAME_SYNC,
-    /// One of the frame parameters was invalid. E.g. the resolution may be too
-    /// small or too large, or the audio sampling rate may be unsupported.
-    TOXAV_ERR_SEND_FRAME_INVALID,
-    /// Either friend turned off audio or video receiving or we turned off
-    /// sending for the said payload.
-    TOXAV_ERR_SEND_FRAME_PAYLOAD_TYPE_DISABLED,
-    /// Failed to push frame through rtp interface.
-    TOXAV_ERR_SEND_FRAME_RTP_FAILED,
-}
+    fn init(&mut self, tox: &mut Tox) {
+        unsafe {
+            let chan: *mut c_void = mem::transmute(&mut *tox.event_tx);
+            ll::toxav_callback_call(self.av, on_call, chan);
+            ll::toxav_callback_call_state(self.av, on_call_state, chan);
+            ll::toxav_callback_bit_rate_status(self.av, on_bit_rate_status, chan);
+            ll::toxav_callback_audio_receive_frame(self.av, on_audio_receive_frame, chan);
+            ll::toxav_callback_video_receive_frame(self.av, on_video_receive_frame, chan);
+        }
+    }
 
-impl Error for TOXAV_ERR_SEND_FRAME {
-    fn description(&self) -> &str {
-        use self::TOXAV_ERR_SEND_FRAME::*;
-        match *self {
-            TOXAV_ERR_SEND_FRAME_OK => "send_frame: no error",
-            TOXAV_ERR_SEND_FRAME_NULL =>
-                "send_frame: one of parameters was null", // FIXME?
-            TOXAV_ERR_SEND_FRAME_FRIEND_NOT_FOUND =>
-                "send_frame: no friend with given friend number",
-            TOXAV_ERR_SEND_FRAME_FRIEND_NOT_IN_CALL =>
-                "send_frame: not in call with friend",
-            TOXAV_ERR_SEND_FRAME_SYNC =>
-                "send_frame: synchronization error occured",
-            TOXAV_ERR_SEND_FRAME_INVALID =>
-                "send_frame: one of parameters was invalid",
-            TOXAV_ERR_SEND_FRAME_PAYLOAD_TYPE_DISABLED =>
-                "send_frame: either we or friend disabled this type of payload",
-            TOXAV_ERR_SEND_FRAME_RTP_FAILED =>
-                "send_frame: failed to push frame through rtp interface",
+    pub fn wait(&self) {
+        unsafe {
+            let delay = ll::toxav_iteration_interval(self.av);
+            sleep(Duration::new(0, delay));
         }
     }
-}
 
-impl fmt::Display for TOXAV_ERR_SEND_FRAME {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.description())
+    pub fn tick(&mut self) {
+        unsafe { ll::toxav_iterate(self.av) };
     }
-}
 
+    pub fn call(
+        &mut self,
+        friend_number: u32,
+        audio_bitrate: u32,
+        video_bitrate: u32
+    ) -> Result<bool, errors::CallError> {
+        Ok(unsafe {
+            tox_try!(err, ll::toxav_call(
+                self.av,
+                friend_number,
+                audio_bitrate,
+                video_bitrate,
+                &mut err
+            ))
+        })
+    }
```

Comment:
Well, I guess that it's fine to keep it that way.. for now `;)` (i.e. I'll probably will make a PR to upstream rstox with formatting "improvements" once this PR will be merged there, and it'll be up to suhr)